dmoj/contrib/coci.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+from dmoj.contrib.default import ContribModule as DefaultContribModule
+from dmoj.error import InternalError
+from dmoj.result import CheckerResult
+from dmoj.utils.helper_files import parse_helper_file_error
+
+
+class ContribModule(DefaultContribModule):
+    AC = 0
+    WA = 1
+    PE = 2
+    IE = 3
+    PARTIAL = 7
+
+    name = 'coci'
+    repartial = re.compile(br'^partial ((\d+)\/(\d*[1-9]\d*))\n$')
+
+    @classmethod
+    def parse_return_code(cls, proc, executor, point_value, time_limit, memory_limit, feedback, name, stderr):
+        if proc.returncode == cls.AC:
+            return CheckerResult(True, point_value, feedback=feedback)
+        elif proc.returncode == cls.PARTIAL:

```

Comment:
Thanks for the patch!

Since these branches look practically the same as the testlib case, could we subclass testlib instead of the default module, and replace all but the branch to `cls.PARTIAL` with a call to `super?`

Reply:
Thanks for the suggestion.

I've updated the COCI ContribModule to extend from Testlib.